File: EdT_ESP2/src/services/authService.ts
import api, { setToken, removeToken } from "./api";
import type {
  LoginRequest,
  AuthResponse,
  UserDto,
} from "../types";

const AUTH_BASE = "/auth";

export const authService = {
  // Connexion
  login: async (data: LoginRequest): Promise<AuthResponse> => {
    const response = await api.post<AuthResponse>(`${AUTH_BASE}/login`, data);
    if (response.data.token) {
      setToken(response.data.token);
    }
    return response.data;
  },

  // Profil connecté
  me: async (): Promise<UserDto> => {
    const response = await api.get<UserDto>(`${AUTH_BASE}/me`);
    return response.data;
  },

  // Changement de mot de passe (premier login)
  changePassword: async (currentPassword: string, newPassword: string): Promise<UserDto> => {
    const response = await api.post<UserDto>(`${AUTH_BASE}/change-password`, {
      currentPassword,
      newPassword,
    });
    return response.data;
  },

  // Logout
  logout: (): void => {
    removeToken();
  },
};

export default authService;

File: edt_mobile/lib/services/auth_service.dart
import 'api_client.dart';
import '../models/user.dart';

class AuthService {
  final _dio = ApiClient().dio;

  /// Login → POST /auth/login
  Future<AuthResponse> login(String email, String password) async {
    final response = await _dio.post('/auth/login', data: {
      'email': email,
      'password': password,
    });
    final auth = AuthResponse.fromJson(response.data);
    // Persist JWT
    await ApiClient.setToken(auth.token);
    return auth;
  }

  /// Get current user profile → GET /auth/me
  Future<UserDto> getMe() async {
    final response = await _dio.get('/auth/me');
    return UserDto.fromJson(response.data);
  }

  /// Logout — clear token
  Future<void> logout() async {
    await ApiClient.removeToken();
  }

  /// Check if authenticated
  Future<bool> isAuthenticated() async {
    return ApiClient.isAuthenticated();
  }
}
